assembl/static2/js/app/searchv1.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import Thunk from 'redux-thunk';
+import { loadTranslations, setLocale, syncTranslationWithStore } from 'react-redux-i18n';
+import RootReducer from './reducers/rootReducer';
+import GlobalFunctions from './utils/globalFunctions';
+import Translations from './utils/translations';
+import { canUseExpertInterface } from './utils/permissions';
+import { fetchPermissionsForUser } from './services/userService';
+
+import { SearchComponent } from './components/search.jsx?v=1'; // eslint-disable-line
+import '../../css/views/searchv1.scss';
+/*
+searchv1.js is included in assembl/templates/index.jinja2
+(you need to change the url for production)
+The search div is included in the page from assembl/templates/views/navBar.tmpl
+and the Search react component is rendered via the onDomRefresh event in
+assembl/static/js/app/views/navBar.js
+*/
+
+
+const myCreateStore = () => {
+  const store = createStore(RootReducer, applyMiddleware(Thunk));
+  const assemblLocale = window.assembl_locale.split('_')[0];
+  const userLocale = GlobalFunctions.getLocale(assemblLocale);
+  syncTranslationWithStore(store);
+  store.dispatch(loadTranslations(Translations));
+  store.dispatch(setLocale(userLocale));
+  return store;
+};
+
+
+class SearchApp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isExpert: false };
+  }
+  componentWillMount() {
+    const discussionId = GlobalFunctions.getDiscussionId();
+    const connectedUserId = GlobalFunctions.getConnectedUserId();
+    this.discussionId = discussionId;
+    this.connectedUserId = connectedUserId;
+    if (connectedUserId) {
+      fetchPermissionsForUser(discussionId, connectedUserId).then((permissions) => {
```

Comment:
This is indeed the case for the v2, where the information is already in the redux store.
For v1, I don't have the info in the store, this is why I do an extra ajax call. But indeed I just saw there is ```<script type="text/json" id="permissions-json">["read", "add_post", "edit_post", "add_extract", "edit_extract", "add_idea", "edit_idea", "edit_synthesis", "send_synthesis", "admin_discussion", "sysadmin", "edit_my_extract", "self_register", "self_register_req", "vote", "read_public_cif", "export_post", "moderate_post", "discussion_stats", "delete_my_post", "delete_post"]</script>``` on the page, so I guess I can use that instead.